magit-file-icons.el

```diff
@@ -55,12 +55,52 @@
   :type 'boolean
   :group 'magit-file-icons)
 
+(defcustom magit-file-icons-icon-backend 'nerd-icons
+  "Icon backend for magit-file-icons. Set this to nil
+if you want to customize `magit-file-icons-icon-for-file-func'
+or `magit-file-icons-icon-for-dir-func'."
+  :type 'symbol
+  :group 'magit-file-icons)
+
+(if (not (require 'nerd-icons nil t)) (funcall
+                                       (lambda ()
+                                        (require 'all-the-icons)
+                                        (setq magit-file-icons-icon-backend 'all-the-icons))))
+
+(defcustom magit-file-icons-icon-for-file-func nil
```

Comment:
Why not use `:type 'function`? It's a perfectly valid `defcustom` type, and means you don't have to explicitly explain how to customize the variable in the docstring.